test/__init__.py

```diff
@@ -673,6 +673,9 @@ def check_auth_with_sharding(self, func):
                              func=func)
 
     def is_topology_type(self, topologies):
+        for t in topologies:
```

Comment:
Why is this needed? Was there an error in the original spec test that this would have helped catch? Can we change this to raise a more informative message when it fails than assertion error? Something like:

```
unsupported_topologies = set(topologies) - {'load-balanced', 'single', 'replicaset', 'sharded', 'sharded-replicaset'}
if unsupported_topologies:
    raise ValueError(f"Specified topology types {unsupported_topologies} are not supported")
```

Reply:
Yes the test accidentally had `["replicaset, sharded"]` instead of `["replicaset", "sharded"]`. Here's an example of the new behavior:
```
ERROR: setUpClass (__main__.TestUnifiedSnapshotSessionsNotSupportedServerError)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 678, in setUpClass
    if not cls.should_run_on(run_on_spec):
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 650, in should_run_on
    if is_run_on_requirement_satisfied(req):
  File "/Users/shane/git/mongo-python-driver/test/unified_format.py", line 103, in is_run_on_requirement_satisfied
    topology_satisfied = client_context.is_topology_type(
  File "/Users/shane/git/mongo-python-driver/test/__init__.py", line 679, in is_topology_type
    raise AssertionError('Unknown topologies: %r' % (unknown,))
AssertionError: Unknown topologies: {'replicaset, sharded'}
```